app/models/tournament.rb

```diff
@@ -21,4 +21,117 @@
 
 class Tournament < Event
   enum game_modes: [:ko, :ko_group, :double_elimination]
+
+  def standing_of(team)
+    last_match = last_match_of team, finale
+    standing_string team, last_match
+  end
+
+  def finale
+    finale = nil
+    matches.each do |match|
+      if match.depth == 0
+        finale = match
+      end
+    end
+    #TODO: keep care of 3rd place match here
+    finale
+  end
+
+  def generate_schedule
+    build_matches filled_teams, max_match_level, 0
+    beautify_match_indices
+  end
+
+  def max_match_level
+    Math.log(teams.length, 2).ceil - 1
+  end
+
+  private
+
+    def last_match_of(team, match)
+      if match.is_team_recursive? team
+        return match
+      end
+      standing_recursion_step(team, match.team_home) || standing_recursion_step(team, match.team_away)
+    end
+
+    def standing_string(team, match)
+      match_name = match.round
+      if match.loser == team
+        return I18n.t 'events.overview.out', match_name: match_name
+      end
+      I18n.t 'events.overview.in', match_name: match_name
+    end
+
+    def standing_recursion_step(team, child)
+      if child.is_a? Match
+        last_match_of team, child
+      end
+    end
+
+    def filled_teams
+      filled_teams = shuffled_teams
+      insert_index = 0
+      until is_power_of_two? filled_teams.length
```

Comment:
We already thought about that. The problem: This would result in uniformly distributed leaf matches, not all packed to the end of the gameday. The POs wanted them packed in the end, so that every team playes at least once before leaf-match-players play again